py/persipubsub/queue.py

```diff
@@ -0,0 +1,489 @@
+#!/usr/bin/env python3
+"""Store messages in a local LMDB."""
+import datetime
+import enum
+import pathlib
+import uuid
+from typing import Any, Dict, List, Optional, Union
+
+import lmdb
+
+import persipubsub
+
+MSG_TIMEOUT_SECS = 500  # type: int
+MAX_MSGS_NUM = 1024 * 64
+HWM_LMDB_SIZE_BYTES = 30 * 1024**3  # type: int
+
+
+class Strategy(enum.Enum):
+    """Hold possible strategies."""
+
+    prune_first = 0
+    prune_last = 1
+
+
+def _parse_strategy(strategy: str) -> Strategy:
+    """
+    Parse overflow strategy.
+
+    :param strategy: Strategy stored in LMDB
+    :return: set overflow strategy
+    """
+    if Strategy.prune_first.name == strategy:
+        return Strategy.prune_first
+
+    if Strategy.prune_last.name == strategy:
+        return Strategy.prune_last
+
+    raise ValueError("Unknown strategy: '{}' choosen.".format(strategy))
+
+
+class HighWaterMark:
+    """Hold high water mark limits."""
+
+    def __init__(self,
+                 msg_timeout_secs: Optional[int] = None,
+                 max_msgs_num: Optional[int] = None,
+                 hwm_lmdb_size_bytes: Optional[int] = None) -> None:
+        """
+        Initialize.
+
+        :param msg_timeout_secs: time after which msg is classified as dangling
+        msg (secs)
+        :param max_msgs_num: maximal amount of msg
+        :param hwm_lmdb_size_bytes: high water mark for total size of LMDB
+        (bytes)
+        """
+        self.msg_timeout_secs = MSG_TIMEOUT_SECS \
```

Comment:
Rename `MSG_TIMEOUT_SECS` to `DEFAULT_MESSAGE_TIMEOUT`? (also the other constants; please don't forget to synchronize the constants with the parameter names)